src/coreclr/jit/optimizer.cpp

```diff
@@ -4201,12 +4201,15 @@ PhaseStatus Compiler::optUnrollLoops()
             // If there is no iteration (totalIter == 0), we will remove the loop body entirely.
             unrollLimitSz = INT_MAX;
         }
-        else if (!(loopFlags & LPFLG_SIMD_LIMIT))
+        else if ((loopFlags & LPFLG_SIMD_LIMIT) == 0)
         {
-            // Otherwise unroll only if limit is Vector_.Length
-            // (as a heuristic, not for correctness/structural reasons)
-            JITDUMP("Failed to unroll loop " FMT_LP ": constant limit isn't Vector<T>.Length (heuristic)\n", lnum);
-            continue;
+            if (loop.lpConstLimit() > opts.compJitUnrollLoopIterationLimit)
```

Comment:
I think you want to be comparing against `totalIter`.

Maybe, for better clarity, the code should be:
```
else if (totalIter <= 1)
{
...
}
else if (totalIter <= opts.compJitUnrollMaxIterationCount)
{
  // We can unroll this
}
else if ((loopFlags & LPFLG_SIMD_LIMIT) != 0)
{
  // We can unroll this
}
else
{
  JITDUMP("Failed to unroll loop " FMT_LP ": insufficiently simple loop (heuristic)\n", lnum);
  continue;
}
```

Reply:
Updated.